src/arch/x86/x86_64/src/bootblock_nomem.S

```diff
@@ -21,15 +21,19 @@
  *
  */
 
+/* XXX: This list required to allow unused named arguments for rustc
+ * {pse} {pge} {pae} {efer} {lme} {cd} {nw} {ts} {mp} {pg} {wp} {pe}
+ * {pte_p} {pte_rw} {pte_ps} {pte2_mpat}
+ * {mtrr_cap_msr} {mtrr_def_type_msr} {mtrr_type_wrback} {mtrr_phys_mask_valid}
+ * {mtrr_def_type_en} {mtrr_type_wrprot}
+ */
```

Comment:
Hmm, it's kinda annoying that this is necessary. Isn't there an option for `global_asm` to turn this off maybe?

If it is necessary, we can make it official by just writing in the above that `the official interface is this: {pse} {pge} ` and so on.

Reply:
Looks like the reason this is needed is because of the `const_asm` macro. For every usage, it's injecting every variable, even if that specific asm file doesn't require it.

It seems fine to:
1. Export constants from the arch crates
2. Use the constants that are required for each `global_asm` invocation as-needed rather than having a unified macro that injects the environment. This seems more readable to me anyway, even if it's more verbose.

Reply:
Indeed, just not having the macro `const_asm` at all would also be a good solution--and just being explicit in what is needed instead, directly using `global_asm`, only with the bindings needed.

The only downside would be that you don't know what the available things in the `.S` file are while you have the `.S` file open in an editor. But those `.S` files hopefully won't need a lot of changes over time anyway--and even if we had the weird comment it would just be a workaround anyway. (in that case, it would still be better to just `include_str` a file which has the correct `global_asm` invocation AND the assembly source code in there)

To summarize, I agree that having `const_asm` automatically bind stuff internally is kinda weird.